plugin/trino-hive/src/main/java/io/trino/plugin/hive/InternalHiveConnectorFactory.java

```diff
@@ -95,7 +95,8 @@ public static Connector createConnector(String catalogName, Map<String, String>
                     new JsonModule(),
                     new TypeDeserializerModule(context.getTypeManager()),
                     new HiveModule(),
-                    new CachingDirectoryListerModule(cachingDirectoryLister),
+                    conditionalModule(HiveConfig.class, hiveConfig -> hiveConfig.isFileStatusCacheTablesEnabled(), new CachingDirectoryListerModule(cachingDirectoryLister)),
+                    conditionalModule(HiveConfig.class, hiveConfig -> !hiveConfig.isFileStatusCacheTablesEnabled(), new DefaultDirectoryListerModule()),
```

Comment:
@findepi IMO it would be better long term to apply similar changes to places where we use caching wrappers which we "disable" by making size and expiry as 0. WDYT?

IMO even though EvictableCache is a useful solution it's easy to not notice places where we have caches susceptible to the synchronisation issues.

Reply:
it's rarely a problem, so i would be weary of a code bloat.

also, you need to do this explicitly, and if you;'re doing this explicitly, providing a no-op cache implementation is much easier (less code) than working through this on guice level.

Reply:
Good point that in order to "fix" existing usages we anyway need to hunt them down.